File: src/js/core/src/Style/MetaSource.ts
import type { StyleDefinitionMetaSource } from '../StyleDefinition.js';

/**
 * Read-only view over the `meta.source` block of a style definition.
 */
export class MetaSource {
  #data: StyleDefinitionMetaSource;

  constructor(data: StyleDefinitionMetaSource) {
    this.#data = data;
  }

  /**
   * Returns the source name (e.g. the original work title), or `undefined`
   * when not set.
   */
  name(): string | undefined {
    return this.#data.name;
  }

  /**
   * Returns the URL of the source, or `undefined` when not set.
   */
  url(): string | undefined {
    return this.#data.url;
  }
}

File: src/dart/core/lib/src/style/meta_source.dart
/// Read-only view over the `meta.source` block of a style definition.
library;

/// The source descriptor of a style's metadata.
///
/// Accessors return `null` for absent fields, distinguishing them from fields
/// explicitly set to the empty string — the license builder relies on that
/// distinction.
class MetaSource {
  final Map<String, Object?> _data;

  MetaSource(Map<String, Object?> data) : _data = data;

  /// Returns the source name (e.g. the original work title), or `null` when
  /// not set.
  String? name() => _data['name'] as String?;

  /// Returns the URL of the source, or `null` when not set.
  String? url() => _data['url'] as String?;
}
